test/unit/mdc-textfield/mdc-text-field-icon.test.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import bel from 'bel';
+import {assert} from 'chai';
+import td from 'testdouble';
+
+import {MDCTextFieldIcon, MDCTextFieldIconFoundation} from '../../../packages/mdc-textfield/icon';
+
+const getFixture = () => bel`
+  <div class="mdc-text-field__icon"></div>
+`;
+
+suite('MDCTextFieldIcon');
+
+test('attachTo returns an MDCTextFieldIcon instance', () => {
+  assert.isOk(MDCTextFieldIcon.attachTo(getFixture()) instanceof MDCTextFieldIcon);
+});
+
+function setupTest() {
+  const root = getFixture();
+  const component = new MDCTextFieldIcon(root);
+  return {root, component};
+}
+
+test('#adapter.setAttr adds a given attribute to the element', () => {
+  const {root, component} = setupTest();
+  component.getDefaultFoundation().adapter_.setAttr('aria-label', 'foo');
+  assert.equal(root.getAttribute('aria-label'), 'foo');
+});
+
+test('#adapter.eventTargetHasClass returns true if classname exists for a given target element', () => {
+  const {root, component} = setupTest();
+  root.classList.add('foo');
+  assert.equal(component.getDefaultFoundation().adapter_.eventTargetHasClass(root, 'foo'), true);
+});
+
+test('#adapter.eventTargetHasClass returns false if classname does not exist for a given target element', () => {
+  const {root, component} = setupTest();
+  assert.equal(component.getDefaultFoundation().adapter_.eventTargetHasClass(root, 'foo'), false);
+});
+
+test('#adapter.notifyIconAction emits ' + `${MDCTextFieldIconFoundation.strings.ICON_EVENT}`, () => {
```

Comment:
Perhaps we should also verify that this event is configured to bubble in this test?